privacyidea/api/lib/postpolicy.py

```diff
@@ -318,6 +319,52 @@ def no_detail_on_success(request, response):
     return response
 
 
+def preferred_client_mode(request, response):
+    """
+    This policy function is used to add the preferred client mode.
+    The admin can set the list of client modes in the policy in the
+    same order as  he preferred them. The faction will pick the first
+    client mode from the list, that is also in the multichallege and
+    set it as preferred client mode
+
+    :param request:
+    :param response:
+    :return:
+    """
+    content = response.json
+    user_object = request.User
+
+    # get the preferred client mode from a policy definition
+    detail_pol = Match.user(g, scope=SCOPE.AUTH, action=ACTION.PREFERREDCLIENTMODE, user_object=user_object)\
+        .action_values(allow_white_space_in_action=True, unique=True)
+
+    if detail_pol:
+        preferred_client_mode_list = list(detail_pol)[0].split(", ")
```

Comment:
This will fail, if the admin enters a policy like `interactive,poll`.
I.e. no whitespace after the comma.

So you should do

~~~~python
preferred_client_mode_list = [cm.strip() for cm inlist(detail_pol)[0].split(",")]
~~~~